File: chapter8_function_call.r
fix_format <- function( text ) {
  text <- gsub( "D", ".", text )
  number <- as.numeric( text )
  return( number )
}

data <- read.csv("broken_data.txt", header = F)

for( i in 1:nrow(data) ){
  salary = data[i, 1]
  months = data[i, 2]
  salary = fix_format( salary )
  months = fix_format( months )
  print( paste( "Annual salary is", salary * months ) )
}

File: chapter8_function_call.py
def fix_format( text ):
  text = text.replace('D', '.')
  number = float( text )
  return number

for line in open('broken_data.txt'):
  salary = line.split(',')[0]
  months = line.split(',')[1]
  salary = fix_format( salary )
  months = fix_format( months )
  print("Annual salary is", salary * months )
